### PR TITLE
aws auth options for aws plugins and nop plugin

### PR DESCRIPTION
**Pull request by boriwo**

### CODE REVIEW

File: pkg/plugins/nop/receiver.go [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+	}
+	// metric recorders
+	meter := global.Meter(rtsemconv.EARSMeterName)
+	commonLabels := []attribute.KeyValue{
+		attribute.String(rtsemconv.EARSPluginTypeLabel, rtsemconv.EARSPluginTypeNopReceiver),
+		attribute.String(rtsemconv.EARSPluginNameLabel, r.Name()),
+		attribute.String(rtsemconv.EARSAppIdLabel, r.tid.AppId),
+		attribute.String(rtsemconv.EARSOrgIdLabel, r.tid.OrgId),
+	}
+	r.eventSuccessCounter = metric.Must(meter).
```
Comment (holidaymike) [inline pkg/plugins/nop/receiver.go:76]: Do we even need metrics for the noop receiver?

File: pkg/plugins/nop/sender.go [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+		NewInt64Histogram(
+			rtsemconv.EARSMetricEventSendOutTime,
+			metric.WithDescription("measures the time ears spends to send an event to a downstream data sink"),
+			metric.WithUnit(unit.Milliseconds),
+		).Bind(commonLabels...)
+	return s, nil
+}
+
+func (s *Sender) Send(e event.Event) {
+	buf, err := json.Marshal(e.Payload())
```
Comment (holidaymike) [inline pkg/plugins/nop/sender.go:104]: is payload in json format a requirement?

Comment (boriwo) [reply pkg/plugins/nop/sender.go:104]: no, it doesn't even get sent, i'm just using this to measure the bytes if it were to be json serialized for metrics purposes

File: pkg/plugins/nop/receiver.go (hunk 1 above)
Comment (boriwo) [reply pkg/plugins/nop/receiver.go]: ok i removed the metrics here

**Pull request merged**